geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/filters/Z3Filter.scala

```diff
@@ -60,20 +123,87 @@ class Z3Filter(
     }
   }
 
+  private def nextJumpIn(epoch: Short, z: Long): FilterResult = {
+    val epochIndex = epoch - epochOffset
+    val zb = if (epochIndex < 0 || epochIndex >= zBounds.length) { zBoundsNoTime } else { zBounds(epochIndex) }
+
+    var nextZ = Long.MaxValue
+
+    var i = 0
+    while (i < zb.bounds.length) {
+      val Array(zmin, zmax) = zb.bounds(i)
+      if (z < zmin) {
+        if (zmin < nextZ) {
+          nextZ = zmin
+        }
+      } else if (z < zmax) {
+        val next = Z3.zdivide(z, zmin, zmax)._2
+        if (next < nextZ) {
+          nextZ = next
+        }
+      }
+      i += 1
+    }
+
+    var nextEpoch = epoch
+
+    if (nextZ == Long.MaxValue && epoch != Short.MaxValue) {
```

Comment:
What's the bit about `epoch != Short.MaxValue`?

Reply:
avoiding numeric wrap-around...